Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM ubuntu
+
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && \
+    apt-get install -y \
```

Comment:
Are all these packages really needed? I think it would be more interesting to keep this list to the bare minimum required, so it reflects the true depedencies for MicroPython, not the complete toolset someone uses. This will also make it easier to keep up to date and in sync with e.g. the Travis configuration.

Reply:
I've only added them because the builds were failing at some point. There's almost nothing in there that I added just to be safe.

Reply:
Ok but how can there be (at first sight) seemingly way more than what's in travis.yml? Thinking of it, would it be possible to generate this list from travis.yml or won't that work because this Dockerfile does more than what Travis does? Basically my number 1 concern for making this part of the main repository is maintainance because it adds extra files which need to be kept up to date with all ports but also with external sources and because it has (I think) duplicate information. Now if this while thing were to be part of the build process, or perhaps maybe the main element of the build, that would be different. And way more interesting. Are there ways to do that?

Reply:
I tried to generate the Dockerfile from travis.yml file, but it the instantly fails, because the base ubuntu image doesn't even ship with add `add-apt-repository`. 
Clearly, travis has a custom build env. 
I tried looking around ([here](https://quay.io/organization/travisci/)), but could not find an ubuntu image.

[Generated Dockerfile and script](https://gist.github.com/devxpy/5fa28b9f73d06d6e3a5124ff7cbd0b77)

I am open to manually "filling in" the needed dependencies if it makes maintenance easier.